src/plugins/discover/public/application/main/components/sidebar/discover_index_pattern.tsx

```diff
@@ -38,16 +38,21 @@ export function DiscoverIndexPattern({
   const options: IndexPatternRef[] = (indexPatternList || []).map((entity) => ({
     id: entity.id,
     title: entity.attributes!.title,
+    readableTitle: entity.attributes?.readableTitle,
   }));
-  const { id: selectedId, title: selectedTitle } = selectedIndexPattern || {};
+  const {
+    id: selectedId,
+    title: selectedTitle,
+    readableTitle: selectedReadableTitle,
+  } = selectedIndexPattern || {};
 
   const [selected, setSelected] = useState({
     id: selectedId,
-    title: selectedTitle || '',
+    title: selectedReadableTitle ? selectedReadableTitle : selectedTitle || '',
   });
   useEffect(() => {
-    const { id, title } = selectedIndexPattern;
-    setSelected({ id, title });
+    const { id, title, readableTitle } = selectedIndexPattern;
+    setSelected({ id, title: readableTitle ? readableTitle : title });
```

Comment:
Does it make sense to provide a method or two on the DataView class that does this little bit of logic? It seems to me there's two use cases. First, when we just want the readable title. - `Sample Data` or `kibana_sample_data`. And then there's the case where we want them combined. Hrm, I'm not sure if I'm accounting for the filter case fully. What do you think?

Reply:
added a getName function to get name if exists otherwise return title